module/extension/Director/src/director/find_ok_solution.cpp

```diff
@@ -0,0 +1,103 @@
+
+#include "Director.hpp"
+
+namespace module::extension {
+
+    using provider::Provider;
+
+    Director::OkSolution Director::find_ok_solution(const Solution& solution,
+                                                    const std::set<std::type_index>& used_types) {
+
+        // We need to accumulate blocked types for options we didn't use
+        // Since we search the options in order
+        std::set<std::type_index> blocking_groups;
+
+        // Look at all the options and build up the SelectedOptions we have
+        for (const auto& option : solution.options) {
+
+            auto o = find_ok_solution(option, used_types);
+
+            // Add any blocked types we found while evaluating this solution
+            blocking_groups.insert(o.blocking_groups.begin(), o.blocking_groups.end());
+
+            // Return the first solution we find that is not blocked
+            if (!o.blocked) {
+                OkSolution(false, o.provider, std::move(o.requirements), std::move(o.used), std::move(blocking_groups));
+                return o;
+            }
+        }
+
+        // We couldn't find an OK solution, return blocked
+        return OkSolution(true, std::move(blocking_groups));
+    }
+
+    Director::OkSolution Director::find_ok_solution(const Solution::Option& option,
```

Comment:
The name `OkSolution` is a meme

Reply:
Think I should change it to `RunnableSolution`?
I wasn't super keen on it either...

Reply:
I like that? Like, ok solution is _ok_ 😝 it just is a bit funny hahaha

Reply:
`Acceptable`? `Adequate`? `Usable`? `Valid`? `Mediocre`?